fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DefaultOCFLObjectSession.java

```diff
@@ -412,14 +415,51 @@ public synchronized void close() throws PersistentStorageException {
     }
 
     @Override
-    public List<VersionDetails> listVersions() throws PersistentStorageException {
+    public List<OCFLVersion> listVersions() throws PersistentStorageException {
         assertSessionOpen();
         //get a list of all versions in the object.
-        return this.ocflRepository.describeObject(this.objectIdentifier)
-                                                                    .getVersionMap().values().stream()
-                                                                    .filter(v -> !v.isMutable())
-                                                                    .sorted(VERSION_COMPARATOR)
-                                                                    .collect(Collectors.toList());
+        try {
+            return this.ocflRepository.describeObject(this.objectIdentifier)
+                    .getVersionMap().values().stream()
+                    .filter(version -> !version.isMutable())
+                    .sorted(VERSION_COMPARATOR)
+                    .map(version -> {
+                        return new OCFLVersionImpl()
```

Comment:
I tend to avoid POJO constructors when possible because they lack clarity and are error prone. Unfortunately, Java doesn't provide any good alternatives that aren't horribly verbose. Recently, I've been favoring setter chaining as a kind of poor man's Builder.